app/src/main/java/com/qudus/tudee/ui/screen/taskEditor/TaskEditorViewModel.kt

```diff
@@ -0,0 +1,88 @@
+package com.qudus.tudee.ui.screen.taskEditor
+
+import com.qudus.tudee.domain.entity.Priority
+import com.qudus.tudee.domain.exception.CategoryFetchAllFailedException
+import com.qudus.tudee.domain.exception.EmptyInputException
+import com.qudus.tudee.domain.exception.InputTooLongException
+import com.qudus.tudee.domain.exception.InputTooShortException
+import com.qudus.tudee.domain.exception.InvalidStartCharacterException
+import com.qudus.tudee.domain.exception.TudeeExecption
+import com.qudus.tudee.ui.base.BaseViewModel
+import com.qudus.tudee.ui.screen.taskEditor.TaskEditorUiState.CategoryItemUiState.Companion.isSameSelection
+import com.qudus.tudee.ui.screen.taskEditor.TaskEditorUiState.CategoryItemUiState.Companion.selectById
+import com.qudus.tudee.ui.screen.taskEditor.TaskEditorUiState.PriorityItemUiState.Companion.isSameSelection
+import com.qudus.tudee.ui.screen.taskEditor.TaskEditorUiState.PriorityItemUiState.Companion.selectByPriorityType
+import kotlinx.coroutines.flow.update
+import kotlinx.datetime.LocalDate
+
+open class TaskEditorViewModel: BaseViewModel<TaskEditorUiState>(TaskEditorUiState()), TaskEditorInteraction {
+
+    override fun onTitleValueChange(newTitle: String) {
+        updateFormState { it.copy(title = newTitle, titleErrorMessageType = null) }
+    }
+
+    override fun onDescriptionValueChange(newDescription: String) {
+        updateFormState { it.copy(description = newDescription) }
+    }
+
+    override fun onDateFieldClick() = _state.update { it.copy(isDatePickerOpen = true) }
+
+    override fun onDateSelected(selectedDate: LocalDate) {
+        _state.update { it.copy(date = selectedDate.toString(), isDatePickerOpen = false) }
```

Comment:
```suggestion
        _state.update { it.copy(date = selectedDate isDatePickerOpen = false) }
```
You can convert it inside mapper or UI

Reply:
I agree that conversion logic should usually go in a mapper, but in this case, there's no need for a separate layer since it's a UI-to-UI transformation. The DatePicker returns a `LocalDate`, but my UI state expects a String, so the conversion is done inline where the interaction happens.